packages/compiler/emitter-framework/types.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  Enum,
+  Interface,
+  Model,
+  ModelProperty,
+  Operation,
+  Program,
+  Scalar,
+  Tuple,
+  Type,
+  Union,
+} from "../core/index.js";
+import { Placeholder } from "./placeholder.js";
+
+export interface AssetEmitter<T> {
```

Comment:
nit: /** */.

Reply:
Leaving many of these docs until later since I want to iterate on them while looking at the actual output